lib/improver/wxcode/weather_symbols.py

```diff
@@ -141,16 +156,28 @@ def check_input_cubes(self, cubes):
                         not self.coord_named_threshold):
                     self.coord_named_threshold = True
 
-                test_condition = (
-                    iris.Constraint(
-                        coord_values={threshold_name: lambda cell: (
-                            threshold * (1. - self.float_tolerance) < cell <
-                            threshold * (1. + self.float_tolerance))},
-                        cube_func=lambda cube: (
-                            find_threshold_coordinate(
-                                cube
-                            ).attributes['spp__relative_to_threshold'] ==
-                            condition)))
+                if threshold == 0.0:
```

Comment:
I think it would be cleaner to build the constraint in one place - the only bit that needs to be in the `if` is the creation of the `coord_values` constraint.